src/util/Xhr.js

```diff
@@ -59,7 +59,11 @@ class Xhr {
         this.axios = axios.create();
 
         if (typeof responseInterceptor === 'function') {
-            this.axios.interceptors.response.use(responseInterceptor);
+            const successInterceptor = responseInterceptor;
+            // Called on any non 2xx response
+            const errorInterceptor: Function = (error: { response: Object }) =>
```

Comment:
I would probably guard and check if there is even a response on the error object. Since in their docs https://github.com/axios/axios#handling-errors they do check if `error.reponse` exists.

BTW that section also shows what could be a request error interceptor use case, aka the request was never made as caught by the lack of `error.response` but existence of `error.request`. Though still not sure for us when that would be the case.

Also, thinking more about it. trying to think what exactly the error interceptor should be returning in its rejected promise to our code after interception has done its job. Maybe we should just return the Error object with new data that may have been modified in the interception to keep the same return if there was no interception.

```js
const errorInterceptor = (error: $AxiosError<any>) => {
    const { response = { data = {} } } = error;
    const { data = {} } = response;
    error.response.data = responseInterceptor(data, error); // the interceptor will modify the data
    return Promise.reject(error); // return the error that came in but with modified data
}
```

In our code, it will hit https://github.com/box/box-ui-elements/blob/master/src/api/Base.js#L157
When there is an error but no interceptor, then the error object is sent in there as-is.
When there is an interceptor, then error object is sent in there with possibility intercepted data.

wdyt?

Reply:
btw can move the function out to the class level too rather having it nested

Reply:
Is there any value to letting developers manipulate the error object before we reject with it? From my understanding people want to intercept errors (read only) so they can do custom stuff based on the status code, but the elements aren't built to render differently if that object is modified right? It might be useful if we supported a custom error message passed into that object.

If we want to allow them to modify the data, we need to document that they have to return something in the error case as well. Or, we could see what their interceptor returns, and if it's undefined we just pass in the unmodified error object. This is starting to feel like a pre-optimization to me.